CHANGELOG.md

```diff
@@ -46,6 +53,39 @@ Released: TBD (Not before 2025-05-01)
   generated for each of these removals.  Like merged class rules above, this
   should only be removing dead code.
   [#594](https://github.com/peggyjs/peggy/pull/594)
+- Character classes now process characters not in the Basic Multi-Lingual
+  Plane (BMP) correctly.  This feature requires a JavaScript environment
+  that supports the `u` flag to regular expressions.  The `u` flag will only
+  be used on character classes that make use of this new feature.
+  [#602](https://github.com/peggyjs/peggy/pull/602)
+- Unicode characters may now be specified with the `\u{hex}` syntax, allowing
+  easier inclusion of characters not in the BMP (such as newer emoji).  This
+  syntax works both in string literals and in character classes.
+  [#602](https://github.com/peggyjs/peggy/pull/602)
+- Errors pointing to non-BMP characters as the "found" text will now show the
+  full character and not the replacement character for the first surrogate in
+  the UTF-16 representation.
```

Comment:
As long as the error doesn't point to the middle of a surrogate pair.  Then you still see the replacement character, and that's probably correct.